contracts/marketplace/direct-listings/DirectListingsLogic.sol

```diff
@@ -469,10 +469,21 @@ contract DirectListingsLogic is IDirectListings, ReentrancyGuardLogic, ERC2771Co
                 IERC1155(_assetContract).balanceOf(_tokenOwner, _tokenId) >= _quantity &&
                 IERC1155(_assetContract).isApprovedForAll(_tokenOwner, market);
         } else if (_tokenType == TokenType.ERC721) {
+            address owner;
+            address operator;
+
+            // failsafe for reverts in case of non-existent tokens
+            try IERC721(_assetContract).ownerOf(_tokenId) returns (address _owner) {
+                owner = _owner;
+
+                try IERC721(_assetContract).getApproved(_tokenId) returns (address _operator) {
```

Comment:
oh is this the call that throws? i would assume that also just returns false if the tokenId doesnt exist?

Reply:
Yeah both `ownerOf` and `getApproved` check if the token-id exists, and revert if not.

Reply:
For most cases, including create and update listing, this is fine. The call would just revert.
But it affects the view function for getting the valid listings.